cmd/main.go

```diff
@@ -8,41 +8,50 @@ import (
 	"github.com/joho/godotenv"
 	"github.com/joshsoftware/profile_builder_backend_go/internal/api"
 	"github.com/joshsoftware/profile_builder_backend_go/internal/app"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/constants"
 	"github.com/joshsoftware/profile_builder_backend_go/internal/repository"
+	"github.com/rs/cors"
+	"go.uber.org/zap"
 )
 
 func main() {
-	ctx := context.Background();
+	ctx := context.Background()
+
+	//setting logger
+	logger, _ := zap.NewProduction()
+	zap.ReplaceGlobals(logger)
+	defer logger.Sync()
 
 	//setup env
-	err := godotenv.Load();
+	err := godotenv.Load()
 	if err != nil {
-		fmt.Println("error loading.env file");
-        return
+		logger.Info("error loading.env file")
+		return
 	}
 
-	fmt.Println("Starting Server...");
-	defer fmt.Println("Shutting Down Server...");
+	fmt.Println("Starting Server...")
+	defer fmt.Println("Shutting Down Server...")
 
 	//Initialize DB
-	db, err := repository.InitializeDatabase(ctx);
-	if err != nil{
-		fmt.Println("Database Error : ", err);
-		return 
+	db, err := repository.InitializeDatabase(ctx)
+	if err != nil {
+		logger.Error("Database Error  : ", zap.Error(err))
+		return
 	}
 	fmt.Println("Connected to Database!")
 
 	//Creating Services
-	services := app.NewServices(db);
+	services := app.NewServices(ctx, db)
 
 	//Initializaing Router
-	router := api.NewRouter(services);
-
-	err = http.ListenAndServe("localhost:1925", router)
-	if err != nil{
-        fmt.Println("Error Starting Server : ", err);
-        return
-    }
+	router := api.NewRouter(ctx, services)
 
+	// CORS middleware
+	cors := cors.New(constants.CorsOptions)
 
-}
\ No newline at end of file
+	err = http.ListenAndServe("localhost:1925", cors.Handler(router))
+	if err != nil {
+		logger.Error("Error Starting Server  : ", zap.Error(err))
+		return
+	}
```

Comment:
Handle graceful shutdowns.
look on this [ref](https://www.reddit.com/r/golang/comments/16l1mhw/looking_for_clarifications_on_how_graceful/)